pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Embeddings
+==========
+
+**Module name:** :mod:`pennylane.templates.embeddings`
+
+.. currentmodule:: pennylane.templates.embeddings
+
+This module provides quantum circuit architectures that can serve as an embedding of inputs
+(represented by the gate parameters) into a quantum state (see also Schuld & Killoran 2019
+:cite:`schuld2019`).
+
+Provided embeddings
+--------------------
+
+Qubit architectures
+*******************
+
+.. autosummary::
+
+    AngleEmbedding
+    AmplitudeEmbedding
+    BasisEmbedding
+
+Continuous-variable architectures
+*********************************
+
+.. autosummary::
+
+    SqueezingEmbedding
+    DisplacementEmbedding
+
+Code details
+^^^^^^^^^^^^
+"""
+#pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane import RX, RY, RZ, BasisState, Squeezing, Displacement, QubitStateVector
+
+
+def AngleEmbedding(features, n_wires, rotation='X'):
+    r"""
+    Uses the entries of ``features`` as rotation angles of qubits.
+
+    The details of the strategy are defined by the `rotation` parameter:
+     * `rotation = 'X'` uses the features to chronologically apply Pauli-X rotations to qubits
+     * `rotation = 'Y'` uses the features to chronologically apply Pauli-Y rotations to qubits
+     * `rotation = 'Z'` uses the features to chronologically apply Pauli-Z rotations to qubits
+
+    The length of ``features`` has to be smaller or equal to the number of qubits. If there are fewer entries in ``features``
+    than rotations, the circuit does not apply the remaining rotation gates.
+
+    This embedding method can also be used to encode a binary sequence into a basis state. Choose rotation='X'
+    and features of a nonzero value of :math:`\pi /2` only where a qubit has to be prepared in state 1.
+
+    Args:
+        features (array): Input array of shape ``(N, )``, where N is the number of input features to embed
+        n_wires (int): Number of qubits in the circuit
```

Comment:
```suggestion
        n_wires (int): Number of wires in the circuit
```
Alternatively, it seems that `AngleEmbedding` could be used on a subset of wires in a QNode, so maybe the docstring should read something like 'number of wires for embedding'?

Reply:
That's a good idea. I rewrote all embeddings to take a list of wires instead of a number of wires, and updated the docs accordingly. This is now consistent with the layers methods. It gets checked if ``wires`` is an iterable.